eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -368,24 +362,14 @@ object Router {
       case Left(lcu) => updateChannelUpdateSameSideAs(lcu.channelUpdate).updateBalances(lcu.commitments)
       case Right(rcu) => updateChannelUpdateSameSideAs(rcu.channelUpdate)
     }
-  }
-  case class LocalChannel(localNodeId: PublicKey, shortChannelId: ShortChannelId, channel: ChannelDetails) {
-    val isPrivate: Boolean = channel match {
-      case _: PrivateChannel => true
-      case _ => false
-    }
-    val capacity: Satoshi = channel.capacity
-    val remoteNodeId: PublicKey = channel match {
-      case c: PrivateChannel => c.remoteNodeId
-      case c: PublicChannel => if (c.ann.nodeId1 == localNodeId) c.ann.nodeId2 else c.ann.nodeId1
-    }
-    /** Our remote peer's channel_update: this is what must be used in invoice routing hints. */
-    val remoteUpdate: Option[ChannelUpdate] = channel match {
-      case c: PrivateChannel => if (remoteNodeId == c.nodeId1) c.update_1_opt else c.update_2_opt
-      case c: PublicChannel => if (remoteNodeId == c.ann.nodeId1) c.update_1_opt else c.update_2_opt
-    }
     /** Create an invoice routing hint from that channel. Note that if the channel is private, the invoice will leak its existence. */
-    def toExtraHop: Option[ExtraHop] = remoteUpdate.map(u => ExtraHop(remoteNodeId, u.shortChannelId, u.feeBaseMsat, u.feeProportionalMillionths, u.cltvExpiryDelta))
+    def toExtraHop: Option[ExtraHop] = {
```

Comment:
nit: I find the name a bit confusing, it doesn't really convey that we'll take the remote direction...maybe `toIncomingExtraHop` would be better?

Reply:
Done in e3f64d2eb.